### PR TITLE
12 get api articles topic

### PR DESCRIPTION
**Pull request by smlbrd**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -134,6 +146,44 @@ describe('GET /api/articles?order', () => {
...
+        });
+      });
+  });
+  test('200: Responds with an empty array if topic exists, but no articles match topic query', () => {
+    return request(app)
+      .get('/api/articles?topic=paper')
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles.length).toBe(0);
+      });
```
Comment (NCDanOlver) [inline __tests__/app.test.js:167]: Really good test, if the topic doesn't exist a valid response is 200 with `[]` as that ___is___ everything that currently matches the topic 🥇

File: __tests__/app.test.js [hunk 2]
```diff
@@ -134,6 +146,44 @@ describe('GET /api/articles?order', () => {
...
+  });
+  test('200: Responds with an array of all articles if query is omitted', () => {
+    return request(app)
+      .get('/api/articles')
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles.length).toBe(5);
+      });
```
Comment (NCDanOlver) [inline __tests__/app.test.js:175]: We already have code to test `GET /api/articles`! We don't need to retest the code here, we're introducing a `topic` query so we should only be testing the behavior of that.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -134,6 +146,44 @@ describe('GET /api/articles?order', () => {
...
+  });
+  test("404: Responds with an error message if topic doesn't exist", () => {
+    return request(app)
+      .get('/api/articles?topic=frogs')
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Not Found');
+      });
```
Comment (NCDanOlver) [inline __tests__/app.test.js:183]: Great test 🥇

File: endpoints.json [hunk 4]
```diff
@@ -106,6 +106,52 @@
...
+        "title": "They're not exactly dogs, are they?",
+        "topic": "mitch",
+        "author": "butter_bridge",
+        "created_at": "2020-06-06T09:10:00.000Z",
+        "votes": 0,
+        "article_img_url": "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        "comment_count": 2
+      }
+    ]
+  },
```
Comment (NCDanOlver) [inline endpoints.json:154]: Nice one! 🥇

File: models/topics.models.js [hunk 5]
```diff
@@ -12,3 +12,18 @@ exports.fetchTopics = () => {
...
+  }
+  return db
+    .query(
+      `SELECT * FROM topics
+    WHERE slug = $1`,
+      [topic]
+    )
+    .then(({ rows }) => {
+      return rows.length > 0 ? true : false;
+    });
```
Comment (NCDanOlver) [inline models/topics.models.js:28]: This is an interesting function for 2 reasons!
- It's a database query that's only returning a boolean?
- I don't think it's actually being invoked anywhere thanks to your solution in the `articles.model`? I did check your `topics.controller` and that only references the `fetchTopics` above.

Does this need to be removed? Thankfully, if you remove it you should relatively easily be able to run all the tests and give yourself confidence it's not a problem (or find out what breaks). The beauty of testing, eh!

**Comment by NCDanOlver** [comment]
Great job with this! 🚀 Be careful not to test tested code, and remove any code that isn't necessary (give it a go and see if something breaks).

I really liked:
- 200 working and tested 🥇 
- Testing 200 for an existing topic that has no articles returning `[]` 🥇 
- 404 for a topic not found 🥇 
- Updating the `endpoints.json`! 🥇

File: models/topics.models.js (hunk 5 above)
Comment (smlbrd) [reply models/topics.models.js:28]: Ah, I can take this out!

It's a vestigial function from when I was trying to do some (what I thought was very clever, but just failed) ternary stuff in my topics.controller to decide whether or not to pass through the topic argument to fetchArticles :joy:

File: __tests__/app.test.js (hunk 2 above)
Comment (smlbrd) [reply __tests__/app.test.js:175]: Understood, thank you! :saluting_face:

**Pull request merged**